PowerEditor/src/MISC/RegExt/regExtDlg.cpp

```diff
@@ -330,47 +334,66 @@ void RegExtDlg::addExt(TCHAR *ext)
 
 		if (dwDisp == REG_OPENED_EXISTING_KEY)
 		{
-			int res = ::RegQueryValueEx(hKey, TEXT(""), nullptr, nullptr, reinterpret_cast<LPBYTE>(valData), reinterpret_cast<LPDWORD>(&valDataLen));
+			long res = ::RegQueryValueEx(hKey, TEXT(""), nullptr, nullptr, reinterpret_cast<LPBYTE>(valData), reinterpret_cast<LPDWORD>(&valDataLen));
 			if (res == ERROR_SUCCESS)
 				::RegSetValueEx(hKey, nppBackup, 0, REG_SZ, reinterpret_cast<LPBYTE>(valData), valDataLen);
 		}
 		::RegSetValueEx(hKey, nullptr, 0, REG_SZ, reinterpret_cast<const BYTE *>(nppName), (lstrlen(nppName) + 1) * sizeof(TCHAR));
 
 		::RegCloseKey(hKey);
 	}
+	else
+	{
+		generic_string error_msg = TEXT("Registring extension failed\r\n\r\n");
+		error_msg += GetLastErrorAsString(nRet);
+		MessageBox(NULL, error_msg.c_str(), NULL, MB_OK | MB_ICONERROR);
+		return false;
+	}
+	return true;
 }
 
 
 bool RegExtDlg::deleteExts(const TCHAR *ext2Delete)
 {
 	HKEY hKey;
-	::RegOpenKeyEx(HKEY_CLASSES_ROOT, ext2Delete, 0, KEY_ALL_ACCESS, &hKey);
+	long  nRet;

```

Comment:
Could be just one line with definition and assignment.

Reply:
Agreed, I just kept the style as used in addExt function. Should both be changed?